### PR TITLE
Validate Document 

### PR DESCRIPTION
**Pull request by TayGov**
# Description

This PR includes the following proposed change(s):

Document validation

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [X] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/extract/DocumentsApiDelegateImpl.java [hunk 1]
```diff
@@ -25,17 +32,30 @@
...
     }
 
     @Override
     public ResponseEntity<DocumentExtractResponse> extractDocumentFormData(UUID xTransactionId, String xDocumentType, MultipartFile file) {
 
         logger.info("document extract request received");
 
+        try {
+            clamAvService.scan(new ByteArrayInputStream(file.getBytes()));
+            if (!TikaAnalysis.isPdf(file)) throw new DiligenDocumentException("Invalid file type");
```
Comment (alexjoybc) [inline src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/extract/DocumentsApiDelegateImpl.java:52]: should that be `InvalidFormatException` something in those lines?

File: src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/extract/DocumentsApiDelegateImpl.java [hunk 2]
```diff
@@ -25,17 +32,30 @@
...
+        } catch (VirusDetectedException e) {
+            throw new AiReviewerVirusFoundException("Virus found in document");
+        } catch (IOException e) {
+            throw new DiligenDocumentException("File is corrupt");
```
Comment (alexjoybc) [inline src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/extract/DocumentsApiDelegateImpl.java:56]: same here

File: src/backend/efiling-reviewer-api/src/test/java/ca/bc/gov/open/jag/efilingreviewerapi/extract/documentsApiDelegateImpl/ExtractDocumentFormDataTest.java [hunk 3]
```diff
@@ -96,4 +108,56 @@ public void withUserHavingValidRequestShouldReturnCredfated() throws IOException
         );
 
     }
+
+    @Test
+    @DisplayName("Error: non pdf throws exception")
+    public void withNonPdfFilesShouldReturnOk() throws IOException, VirusDetectedException {
+
+        Mockito.when(diligenService.postDocument(Mockito.any(), Mockito.any())).thenReturn(BigDecimal.ONE);
```
Comment (alexjoybc) [inline src/backend/efiling-reviewer-api/src/test/java/ca/bc/gov/open/jag/efilingreviewerapi/extract/documentsApiDelegateImpl/ExtractDocumentFormDataTest.java:116]: We are duplicating this mock, I think it's better if most of them are in the `beforeall` method ? thoughts ?

**Review by alexjoybc — CHANGES_REQUESTED** [review]
Question around the usage of an exception, looks good otherwise

**Review by akroon3r — APPROVED** [review]
Looking real 🐦

**Pull request merged**